server/idporten.ts

```diff
@@ -1,34 +1,49 @@
 import fetch from "node-fetch";
-import {createRemoteJWKSet, jwtVerify} from "jose";
-import {Issuer} from "openid-client";
+import {
+  createRemoteJWKSet,
+  FlattenedJWSInput,
+  JWSHeaderParameters,
+  jwtVerify,
+  KeyLike,
+} from "jose";
+import { Issuer } from "openid-client";
+import { GetKeyFunction } from "jose/dist/types/types";
 
-const acceptedAcrLevel = 'Level4';
-const acceptedSigningAlgorithm = 'RS256';
+const acceptedAcrLevel = "Level4";
+const acceptedSigningAlgorithm = "RS256";
 
-let idportenIssuer;
-let _remoteJWKSet;
+let idportenIssuer: Issuer;
+let _remoteJWKSet:
+  | KeyLike
+  | Uint8Array
+  | GetKeyFunction<JWSHeaderParameters, FlattenedJWSInput>;
 
 //const IDPORTEN_WELL_KNOWN_URL="https://fakedings.dev-gcp.nais.io/fake/.well-known/openid-configuration"
 
 export async function initIdporten() {
-    if (process.env.NODE_ENV === 'labs-gcp') {
-        // I labs så returnerer vi mock uansett
-        return;
-    }
-    idportenIssuer = await Issuer.discover(process.env.IDPORTEN_WELL_KNOWN_URL);
-    _remoteJWKSet = createRemoteJWKSet(new URL(idportenIssuer.metadata.jwks_uri));
+  if (process.env.NODE_ENV === "development") {
+    // I labs så returnerer vi mock uansett
+    return;
+  }
+  idportenIssuer = await Issuer.discover(process.env.IDPORTEN_WELL_KNOWN_URL!);
+  _remoteJWKSet = createRemoteJWKSet(
+    new URL(idportenIssuer.metadata.jwks_uri!)
+  );
 }
 
-export async function verifiserAccessToken(token) {
-    const {payload} = await jwtVerify(token, _remoteJWKSet, {
-        algorithms: [acceptedSigningAlgorithm],
-        issuer: idportenIssuer.metadata.issuer,
-    });
+export async function verifiserAccessToken(token: any) {
+  // @ts-ignore
+  const { payload } = await jwtVerify(token, _remoteJWKSet, {
+    algorithms: [acceptedSigningAlgorithm],
+    issuer: idportenIssuer.metadata.issuer,
+  });
```

Comment:
Her bør man også sjekke at `client_id` claimet er lik miljøvariabelen `IDPORTEN_CLIENT_ID` - ellers så åpner man i teorien for at tokens tilhørende andre apper/tjenester I ID-porten blir godkjent.

Noe sånt kanskje:

```js
assert.equal(jwtToken.payload['client_id'], process.env.IDPORTEN_CLIENT_ID)
```

Reply:
Vi vil jo at andre nav apper er godkjent, eller gjelder det du snakker om apper ut fra nav?

Reply:
Begge deler. Zero trust handler om at du kun stoler på tokens som er ment for din app. Det gir ikke så mye mening at naboen din sine nøkler fungerer på din lås bare fordi de har samme produsent.

Reply:
Men hva skjer om brukeren er innlogget i en app og beveger seg fra den ene til den andre. 
Eksempel typisk reise hos oss i PO arbeidsgiver, en bruker logger seg inn hos oss og går deretter til `min-side-arbeisgiver` , betyr det at brukeren må logge seg på nytt eller?

Reply:
Hvis brukeren har vært innom ID-porten nylig så har vedkommende en single sign-on (SSO) sesjon hos dem. Når brukeren går fra dere til `min-side-arbeidsgiver` så blir de først sendt til ID-porten for å sjekke at SSO-sesjonen fremdeles er gyldig. Hvis den er det blir brukeren sendt rett tilbake med en gang uten å måtte logge seg inn på nytt.

Reply:
Gode innspill her. Tar disse med oss :)